File: ex_prime_fact.py
def prime_factorization(num):
  print(f"{num} = ", end="")
  i=2
  while(i<=num):
    if(num==i):
      print(f"{i}\n", end="")
      return 
      
    if(num%i==0):
      print(f"{i} * ", end="")
      num/=i
      
    else:
      i+=1
      
    
  
if __name__ == "__main__":
  prime_factorization(462)
  

File: ex_prime_fact.c
#include <stdio.h>

void prime_factorization(int num) {
  printf("%d = ", num);
  int i = 2;
  while (i <= num) {
    if (num == i) {
      printf("%d\n", i);
      return;
    }
    if (num % i == 0) {
      printf("%d * ", i);
      num /= i;
    } else {
      i += 1;
    }
  }
}

int main() {
  prime_factorization(462);
}